calabash-cucumber/lib/calabash-cucumber/launch/simulator_launcher.rb

```diff
@@ -251,13 +251,26 @@ def app_bundle_or_raise(path=nil, device_build_dir='iPhoneSimulator')
             puts('-'*37)
           end
         else
-          bo_dir = build_output_dir_for_project
-          sim_dirs = Dir.glob(File.join(bo_dir, '*-iphonesimulator', '*.app'))
+          search_dir = build_output_dir_for_project || DERIVED_DATA
+          sim_dirs = ''
+          apps = `find #{search_dir} -type d -name "*.app" -exec stat -f "%m %N" {} \\; | sort -rn | cut -d" " -f2`.split("\n")
+          apps.each do |app_path|
+            lipo = RunLoop::Lipo.new(app_path)
+            arches = lipo.info
+            if arches.include?("x86_64") || arches.include?("i386")
+              app =  RunLoop::App.new(app_path)
+              executable_name = app.executable_name
+              path_to_bin = File.join(app_path, executable_name)
+              if `xcrun strings "#{path_to_bin}" | grep -E 'CALABASH VERSION'`.include? "CALABASH VERSION"
+                sim_dirs = Dir.glob(app_path)
+              end
+            end
+          end
```

Comment:
In the worst case, this looks like it will iterate over _all_ apps that found and then `sim_dirs` will be the last app that matches the criteria and not the first.

I think we want to use:  `apps.find do` instead of `apps.each`

```
sim_dirs = apps.find do |app_path|

end
```